apps/src/applab/applab.js

```diff
@@ -1306,6 +1309,17 @@ Applab.onPropertyChange = function(element, name, value) {
       element.style.backgroundImage = 'url(' + value + ')';
       element.style.backgroundSize = width + 'px ' + height + 'px';
       break;
+    case 'picture':
+      element.src = value;
+      element.onload = function () {
+        element.style.width = this.naturalWidth + 'px';
```

Comment:
I went on a wild goose chase trying to figure out what "this" is here, before realizing that it refers to the element. It is a little confusing to refer to it as 'element' and 'this' in the same line, however I think it makes sense because in this context 'element' is the model we are updating and 'this' is the HTML element we're pulling a property off of. All this to say, can you maybe just add a comment saying 'this and element happen to be the same object here' or similar? 

Tangentially, this (bad pun intended) reminds me that onPropertyChange is a great candidate to (at some point) move into an ElementProperties base class which ButtonProperties, LabelProperties, etc inherit from. This would make it easier to override property-setting behavior on a per-property, per-element basis.